src/services/ndjson.service.js

```diff
@@ -0,0 +1,24 @@
+const fs = require('fs');
+
+/**
+ * Retrieves ndjson content from the specified url and returns it in the
+ * response.
+ * @param {*} request the request object passed in by the user
+ * @param {*} reply the response object
+ */
+const returnNDJsonContent = async (request, reply) => {
+  request.log.info('Base >>> return NDJSON');
+  const clientId = request.params.clientId;
+  const url = request.params.fileName;
+  const filePath = `tmp/${clientId}/${url}`;
+  if (fs.existsSync(filePath)) {
+    const readStream = fs.createReadStream(`tmp/${clientId}/${url}`);
+    readStream.on('data', function (text) {
+      reply.header('Content-type', 'application/ndjson+fhir').send(text);
+    });
```

Comment:
This current pattern logs a fastify error to the console:

![image](https://user-images.githubusercontent.com/16297930/141497797-16d88ec6-75c5-4c75-adad-0abe2af2c24a.png)

I think this is telling us that the function sends a void response before processing the stream. I read online that you can send the stream directly back with the reply rather than using the `on('data')` hook and it will work exactly the same, but this error no longer happens.

```suggestion
    const readStream = fs.createReadStream(`tmp/${clientId}/${url}`);
    reply.header('Content-type', 'application/ndjson+fhir').send(readStream);
```